### PR TITLE
Add info on lack of deep observability

### PR DESCRIPTION
**Pull request by MaikuB**
I also removed a period that i missed in #243 and updated the info about the MobX quick start guide to say nearly all of the concepts are applicable instead of having it say it's 100% applicable given deep observability isn't possible

### CODE REVIEW

File: docs/src/core-concepts.mdx [hunk 1]
```diff
@@ -75,6 +75,9 @@ will fade away and you will mostly focus on the code within the braces.
 
 The above code is part of the [**Counter Example**](/examples/counter).
 
+For those who have experience with the JavaScript version of MobX, it's important to note that deep observability doesn't apply in the Dart version of MobX. This is because Dart is a statically typed language and there's no support for reflection in Flutter applications. Therefore, if a complex object is marked as observable then
+MobX will not be able to automatically track changes to the object's properties. Should change tracking need to enabled on the object's properties then the properties need to be marked as observable as well.
```
Comment (pavanpodila) [inline docs/src/core-concepts.mdx]: > Should change tracking need to enabled on the object's properties then the properties need to be marked as observable as well.

I think this sentence should be worded differently.

Comment (MaikuB) [reply docs/src/core-concepts.mdx]: What's the reason behind your thought and what suggestions do you have? I could change it to "Should change tracking need to enabled on the object's properties then they should be marked as observable as well."

Comment (pavanpodila) [reply docs/src/core-concepts.mdx]: I was thinking: "Should you need change tracking at the field level, its better to mark them with `@observable`"

Comment (MaikuB) [reply docs/src/core-concepts.mdx]: done, with a minor tweak to that sentence

**Pull request merged**